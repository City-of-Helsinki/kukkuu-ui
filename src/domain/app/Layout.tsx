import 'react-toastify/dist/ReactToastify.css';
import { Outlet } from 'react-router';
import { useSelector } from 'react-redux';

import { isSessionExpiredPromptOpenSelector } from './state/ui/UISelectors';
import SessionAlert from './sessionAlert/SessionAlert';
import PageLayout from './layout/PageLayout';
import useHashAnchorLinks from './useHashAnchorLinks';
import AppTitleAnnouncer from './AppTitleAnnouncer';
import { ScrollToTop } from '../../common/route/RouteUtils';

const Layout = () => {
  useHashAnchorLinks();

  const isSessionPromptOpen = useSelector(isSessionExpiredPromptOpenSelector);
  return (
    <>
      <ScrollToTop />
      <PageLayout>
        <AppTitleAnnouncer />
        {isSessionPromptOpen && <SessionAlert isOpen={isSessionPromptOpen} />}
        <Outlet />
      </PageLayout>
    </>
  );
};

export default Layout;
