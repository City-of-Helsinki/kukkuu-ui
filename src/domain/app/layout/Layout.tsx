import React, { FunctionComponent } from 'react';

import Header from '../header/Header';
import styles from './layout.module.scss';
import Footer from '../footer/Footer';
import Container from './Container';

const PageLayout: FunctionComponent = ({ children }) => {
  return (
    <div className={styles.pageWrapper}>
      <Header />

      <div className={styles.pageBody}>
        <Container>{children}</Container>
      </div>

      <Footer />
    </div>
  );
};

export default PageLayout;
