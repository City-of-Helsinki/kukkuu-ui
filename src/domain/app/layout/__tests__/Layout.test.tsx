import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Layout from '../Layout';

it('renders snapshot correctly', () => {
  const layout = shallow(
    <Layout>
      <></>
    </Layout>
  );
  expect(toJson(layout)).toMatchSnapshot();
});
