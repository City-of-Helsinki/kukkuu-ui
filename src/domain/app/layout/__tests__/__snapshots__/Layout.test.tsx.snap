// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders snapshot correctly 1`] = `
<div
  className="pageWrapper"
>
  <Header />
  <div
    className="pageBody"
  />
  <Footer />
</div>
`;
